src/views/DataDownloadView.vue

```diff
@@ -0,0 +1,191 @@
+    this.$watch(
+      () => this.$route.params,
+      () => {
+        this.fetchPanels();
+      },
+      // fetch the data when the view is created and the data is
+      // already being observed
+      { immediate: true }
+    );
+  },
+  methods: {
+    fetchPanels() {
+      this.panelErrorMsg = this.panelData = null;
+      this.isDataLoading = true;
+      fetch("/gene2phenotype/api/panels/")
+        .then((response) => {
+          if (response.status === 200) {
+            return response.json();
+          } else {
+            return Promise.reject(
+              new Error("Unable to fetch panels data. Please try again later.")
+            );
+          }
+        })
+        .then((responseJson) => {
+          this.isDataLoading = false;
+          this.panelData = responseJson;
+        })
+        .catch((error) => {
+          this.isDataLoading = false;
+          this.panelErrorMsg = error.message;
+          console.log(error);
+        });
+    },
+    downloadPanelData(panelName) {
+      let responseContentDisposition = null;
+      this.dataDownloadErrorMsg = null;
+      this.isDataLoading = true;
+      fetch(`/gene2phenotype/api/panel/${panelName}/download`, {
+        method: "GET",
+        headers: {
+          "content-type": "text/csv;charset=UTF-8",
+        },
+      })
+        .then((response) => {
+          responseContentDisposition = response.headers.get(
+            "Content-Disposition"
+          );
+          if (response.status === 200) {
+            return response.text();
+          } else {
+            return Promise.reject(
+              new Error("Unable to download data. Please try again later.")
+            );
+          }
+        })
+        .then((responseText) => {
+          this.isDataLoading = false;
+          // get csv file name from response Content-Disposition header
+          const regexMatch = responseContentDisposition.match(
+            /attachment; filename="([^"]+)"/
+          ); // Ex responseContentDisposition value: attachment; filename="some_file_name.csv"
+          let csvFileName = "data.csv"; //default csv file name
+          if (regexMatch && regexMatch.length > 0 && regexMatch[1]) {
+            csvFileName = regexMatch[1];
+          }
+          // download csv data to file
+          const csvDataText = responseText;
+          const anchor = document.createElement("a");
+          anchor.href =
+            "data:text/csv;charset=utf-8," + encodeURIComponent(csvDataText);
+          anchor.target = "_blank";
+          anchor.download = csvFileName;
+          anchor.click();
+        })
+        .catch((error) => {
+          this.isDataLoading = false;
+          this.dataDownloadErrorMsg = error.message;
+          console.log(error);
+        });
+    },
+  },
+};
+</script>
+<template>
+  <div class="container px-5 py-3" style="min-height: 60vh">
+    <h2 class="pb-2">Download G2P data by panel</h2>
```

Comment:
On the main page we call it `Disorder panel`.
This page could be renamed to `Download G2P data by disorder panel`.

Reply:
taken care in latest commit